dask/array/core.py

```diff
@@ -1831,18 +1839,27 @@ def __abs__(self):
         return elemwise(operator.abs, self)
 
     def __add__(self, other):
+        from ..dataframe import DataFrame, Series
+        if isinstance(other, (DataFrame, Series)):
+            return NotImplemented
```

Comment:
This helps dask, pandas and numpy negotiate which __add__ method handles the logic of adding to one of the other types.